### PR TITLE
Issue #370 second fix teams controller

### LINKED ISSUE #370: Second Fix TeamsController
closes #360

### PR DESCRIPTION
**Pull request by YuriiChornyi**
### Summary

changed controller and few things on service and repository level

closes #370 

### Areas Of Interest

_Any specific places you want to bring attention of the reviewers? Some places you are uncertain in/some design decisions. (If none remove this section)_

#### Design

- [ ] Web API layer contains as little logic as possible
- [ ] Domain objects use semantic names

#### Perfromance

- [ ] In case of any DB query was changed: attach generated SQL for review
- [ ] Number of DB roundtrips should be as low as possible

#### Unit Tests

- [ ] Naming: Initial state and expected result are properly stated
- [ ] Test follows [AAA](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80) pattern.
- [ ] Test is Isolated
- [ ] Mock instances and expected instances do not share references
- [ ] When new property is added all related comparers are updated

### CODE REVIEW

File: src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Queries/TeamQueries.cs [hunk 1]
```diff
@@ -142,12 +142,14 @@ public Team Execute(FindByNameCriteria criteria)
 
         private static Team GetTeamMapping(TeamEntity t)
         {
+            var roster = t.Players?.Select(p => new PlayerId(p.Id)) ?? new List<PlayerId> { new PlayerId(t.CaptainId) };
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Queries/TeamQueries.cs]: Why do you need this?

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs [hunk 2]
```diff
@@ -102,24 +100,39 @@ public JsonResult Create(TeamViewModel teamViewModel)
...
+                        var team = teamViewModel.ToCreateTeamDto();
+                        var createdTeam = _teamService.Create(team);
+                        teamViewModel.Id = createdTeam.Id;
+                    }
+                    else
+                    {
+                        var team = teamViewModel.ToCreateTeamDto();
+                        var createdTeam = _teamService.Create(team);
+                        teamViewModel.Id = createdTeam.Id;
+                        createdTeam.SetCaptain(new PlayerId(teamViewModel.Captain.Id));
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs]: Why do you need this call? Isn't captain already set?

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs [hunk 3]
```diff
@@ -102,24 +100,39 @@ public JsonResult Create(TeamViewModel teamViewModel)
...
+                    }
+                    
+                    if (teamViewModel.AddedPlayers.Count > 0)
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs]: What about removed players?

Comment (YuriiChornyi) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs]: it is create controller, probably there is no deleted players

File: src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Queries/TeamQueries.cs (hunk 1 above)
Comment (YuriiChornyi) [reply src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Queries/TeamQueries.cs]: I changed this, problem was in nullrefexception

**Pull request merged**